### PR TITLE
feat: Add support for list of dropout rates in MLP model

### PR DESCRIPTION
**Pull request by bagustris**
This commit enhances the MLP model to support a list of dropout rates for different layers (fix issue in [bagustris/issues/7](https://github.com/bagustris/nkululeko/issues/7)).

Previously, the dropout rate was a single float value applied to all layers. This change allows for more granular control over regularization by specifying different dropout rates for each layer.

The changes include:
- Modified `MLPModel` to parse a list of dropout rates from the configuration.
- Updated the `MLP` inner class to apply the dropout rates from the list to the corresponding layers.
- Added a new test case to verify the new functionality.

I tried it works with the following INI file (see drop values):

```ini
[EXP]
root = ./examples/results/
name = exp_polish_mlp_drops
runs = 1
epochs = 5

[DATA]
databases = ['train', 'dev', 'test']
train = ./data/polish/polish_train.csv
train.type = csv
train.absolute_path = False
train.split_strategy = train
dev = ./data/polish/polish_dev.csv
dev.type = csv
dev.absolute_path = False
dev.split_strategy = train
test = ./data/polish/polish_test.csv
test.type = csv
test.absolute_path = False
test.split_strategy = test
target = emotion
labels = ['anger', 'neutral', 'fear']

[FEATS]
type = ['os']

[MODEL]
type = mlp
layers = [64, 32]
drop = [0.2, 0.1]

[RESAMPLE]
replace = True
```

### CODE REVIEW

File: ini_file.md [hunk 1]
```diff
@@ -473,7 +473,8 @@ Model and training specifications. In general, default values should work for cl
 * **learning_rate**: The learning rate for ANN models
   * learning_rate = 0.0001
 * **drop**: Adding dropout (after each hidden layer). Value states dropout probability
-  * drop = .5
+  * drop = .5 # or [0.2] to apply dropout rate 0.5 for all layers or a single layer
```
[the remark this answers is not in the document]
Comment (bagustris) [reply ini_file.md:476]: No, this is the current behavior. If the same dropout rate will be applied on multi layers or on a single layer.
